### PR TITLE
distro-features: Add displbe & sndbe

### PR DESCRIPTION
**Pull request by svlad-90**
## distro-features: Add displbe & sndbe

In case of virtio functionality, we do not need installation of the
displbe and sndbe services. Driver domain should not install and
execute those services. Control domain should not have dependencies to
those services.

This patch is intended to introduce 'displbe' and 'sndbe' ditribution
features, which allow to turn mentioned services on and off for a
specific product.

## virtio: Introduce ENABLE_VIRTIO moulin parameter

We need a separate moulin parameter to implement builds with the virtio
devices sharing capabilities.

This patch adds first iteration of such a parameter. As of now, it does
only turn off the displbe and sndbe distro features.

## readme: Introduce table of contents

It is much easier to navigate over the documentation, when it has table
of contents. This patch creates such a table in the README.md

### CODE REVIEW

**Comment by svlad-90** [comment]
The change was verified in the following way:

|Build type|displbe & sndbe status|Link to the graph|
|--MACHINE h3ulcb-4x2g-kf --ENABLE_ANDROID yes --ENABLE_MM yes --ENABLE_VIRTIO yes|displbe & sndbe are NOT presented in the Yocto graph|Empty graph|
|--MACHINE h3ulcb-4x2g-kf --ENABLE_ANDROID yes --ENABLE_MM yes --ENABLE_VIRTIO no|displbe & sndbe are presented in the Yocto graph|[displbe.txt](https://github.com/xen-troops/meta-xt-prod-devel-rcar/files/11773706/displbe.txt), [sndbe.txt](https://github.com/xen-troops/meta-xt-prod-devel-rcar/files/11774562/sndbe.txt)|
|--MACHINE h3ulcb-4x2g-kf --ENABLE_ANDROID yes --ENABLE_MM yes|displbe & sndbe are presented in the Yocto graph|[displbe.txt](https://github.com/xen-troops/meta-xt-prod-devel-rcar/files/11773706/displbe.txt), [sndbe.txt](https://github.com/xen-troops/meta-xt-prod-devel-rcar/files/11774562/sndbe.txt)|

File: README.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # meta-xt-prod-devel-rcar #
 
+# Table of contents
```
Comment (lorc) [inline README.md:3]: This is a good change, but it should go into a separate commit

**Review by lorc — COMMENTED** [review]
Thank you for the changes. I am generally fine with the PR, but you need to re-arrange some changes.

Generally we want functional changes and corresponding documentation changes to be in the same commit to preserve atomicity. Also, this is a good way to document patch itself.

One more note - "Extension of the README.md" - is to generic. It tells almost nothing about the commit

File: meta-xt-prod-devel-rcar-control/recipes-guests/doma/doma.bbappend [hunk 2]
```diff
@@ -24,7 +26,11 @@ do_install:append() {
     install -d ${D}${sysconfdir}/systemd/system/doma.service.d
     install -m 0644 ${WORKDIR}/doma-set-root.conf ${D}${sysconfdir}/systemd/system/doma.service.d
 
-    # Install drop-in file to add dependencies on sndbe and displbe
```
[the remark this answers is not in the document]
Comment (svlad-90) [reply meta-xt-prod-devel-rcar-control/recipes-guests/doma/doma.bbappend:27]: Hi @lorc, done.

File: prod-devel-rcar-agl.yaml [hunk 3]
```diff
@@ -765,3 +765,21 @@ parameters:
             builder:
               env:
                 - "DDK_UM_PREBUILDS=eva/pvr-um"
+
+  ENABLE_VIRTIO:
```
[the remark this answers is not in the document]
Comment (svlad-90) [reply prod-devel-rcar-agl.yaml:772]: Hi @lorc, done.

File: README.md (hunk 1 above)
Comment (svlad-90) [reply README.md:3]: Hi @lorc, done.

**Comment by svlad-90** [comment]
Hi @lorc, all requested changes were applied. Please, have a look.

**Review by lorc — APPROVED** [review]
Thank you

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by svlad-90** [comment]
Hi @lorc, Reviewed-by was added to all the commits. Please, merge this PR, when suitable.
